samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+The forwarding/processing decision is made in the same way as in the *Header inspection* step.
+
+NOTE: `PartitionMappingFailedException` is considered an [unrecoverable exception](/nservicebus/recoverability/custom-recoverability-policy.md). Whenever such an exception is raised the message that triggered the exception will be moved immediately to the configured error queue.
+
+### Control message forwarding
+
+When an endpoint instance receives a control message representing either the [Subscribe or Unsubscribe intent](/nservicebus/messaging/headers.md#messaging-interaction-headers-nservicebus-messageintent), the message is forwarded to all other partitions.
+
+### Configuration
+
+To enable Receiver Side Distribution for a specific endpoint, two arguments need to be provided:
+- Endpoint discriminators that are based on Service Fabric partition keys 
+- Mapping function per incoming message type that maps incoming messages to a partition key
+
+Use this code to enable Receiver-Side distribution:
+
+snippet: ConfigureReceiverSideDistribution-CandidateVoteCount
+
+In many low-throughput scenarios Receiver Side Distribution might be enough to get started. To achieve better performance, it would be more efficient to route directly to the partitions from the sender with *Sender-Side Distribution*.
+
+## Sender Side Distribution
+
+Receiver Side Distribution addresses forwarding messages that arrive to the incorrect partition. Forwarding received messages introduces some overhead though. To remove the overhead on the receiver side the Sender Side Distribution approach can be used to distribute messages to the correct endpoint instances based on Service Fabric partitioning information.
+
+Sender Side Distribution can be applied to endpoints hosted inside Service Fabric by using the partition information of the stateful services. This is suitable for endpoints hosted inside the cluster that need to send messages to other endpoints hosted in the cluster. For the endpoints hosted outside of the cluster access to Service Fabric APIs is not possible. Instead, partitioned destination endpoints need to be registered with message mapping for commands sent out.
+
+The Sender Side Distribution works in the following way:
+
+1. A mapping function is applied when dispatching messages. This mapping function is intended to select a partition key based on business criteria. In this example it's either the candidate name or the zip code of the voter, depending on the message type and destination endpoint.
+1. The result of this mapping, a partition key, is then added as a `partition-key` header to the outgoing message. This ensures that its value doesn't have to be calculated on the receiver side again and no receiver side distribution will occur. 
+1. With determined destination, the message is sent to the instance specific queue directly.
```

Comment:
@ramonsmits I think we can make that assumption in the context on NServiceBus.

As a context information: inside SF each partition has number of replicas (exact number is a configuration thing and defaults to 3 when there is no failover happening) - one being the primary. The default configuration is that only singe instance of a process will be run for each primary replica. It is possible to run more than one process/instance per replica. In such case the secondaries are treated as read-only. But as I said I don't think this applies to NServiceBus - we don't have a concept of read-only endpoint instance.